Non AWS ProviderID causes CCM to Panic

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!-->


**What happened**:
I have mixed nodes, some on AWS and some not. AWS CCM crashes when a providerID comes up that is not in the right format

**What you expected to happen**:
AWS CCM should ignore a node that doesn't have an `aws:///` providerID

**How to reproduce it (as minimally and precisely as possible)**:
I'm using K3s. I have the CCM Helm chart installed but it's basically just running the ds with these flags:

 --configure-cloud-routes=false --v=2 --cloud-provider=aws

When it gets to a node that isn't aws:/// it does this:

```
E0715 23:48:08.031944       1 runtime.go:79] Observed a panic: &errors.errorString{s:"unable to calculate an index entry for key \"147-185-40-113\" on index \"instanceID\": error mapping node \"147-185-40-113\"'s provider ID \"not:///asdf/asdf\" to instance ID: Invalid format for AWS instance (not:///asdf/asdf)"} (unable to calculate an index entry for key "147-185-40-113" on index "instanceID": error mapping node "147-185-40-113"'s provider ID "not:///asdf/asdf" to instance ID: Invalid format for AWS instance (not:///asdf/asdf))
goroutine 355 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x25e3d20?, 0xc000f38170})
	k8s.io/apimachinery@v0.27.0/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xfffffffe?})
	k8s.io/apimachinery@v0.27.0/pkg/util/runtime/runtime.go:49 +0x75
panic({0x25e3d20, 0xc000f38170})
	runtime/panic.go:884 +0x213
k8s.io/client-go/tools/cache.(*storeIndex).updateIndices(0xc000b882a0, {0x0, 0x0}, {0x2aa73c0?, 0xc000f2e000}, {0xc000cb37c0, 0xe})
	k8s.io/client-go@v0.27.0/tools/cache/thread_safe_store.go:162 +0x6b3
k8s.io/client-go/tools/cache.(*threadSafeMap).Update(0xc000b24ea0, {0xc000cb37c0, 0xe}, {0x2aa73c0?, 0xc000f2e000})
	k8s.io/client-go@v0.27.0/tools/cache/thread_safe_store.go:226 +0x14b
k8s.io/client-go/tools/cache.(*threadSafeMap).Add(0x2aa73c0?, {0xc000cb37c0?, 0x2aa73c0?}, {0x2aa73c0?, 0xc000f2e000?})
	k8s.io/client-go@v0.27.0/tools/cache/thread_safe_store.go:218 +0x28
k8s.io/client-go/tools/cache.(*cache).Add(0xc000a9aa50, {0x2aa73c0, 0xc000f2e000})
	k8s.io/client-go@v0.27.0/tools/cache/store.go:155 +0xb0
k8s.io/client-go/tools/cache.processDeltas({0x308d1e0, 0xc000add4a0}, {0x309a860, 0xc000a9aa50}, {0xc000cc8600?, 0x1, 0xc000c4d1e0?}, 0x28?)
	k8s.io/client-go@v0.27.0/tools/cache/controller.go:459 +0x1c9
k8s.io/client-go/tools/cache.(*sharedIndexInformer).HandleDeltas(0xc000add4a0, {0x269f960?, 0xc000ede960?}, 0x1?)
	k8s.io/client-go@v0.27.0/tools/cache/shared_informer.go:638 +0x176
k8s.io/client-go/tools/cache.(*DeltaFIFO).Pop(0xc000a7a2c0, 0xc0008d9950)
	k8s.io/client-go@v0.27.0/tools/cache/delta_fifo.go:603 +0x5be
k8s.io/client-go/tools/cache.(*controller).processLoop(0xc0008e0640)
	k8s.io/client-go@v0.27.0/tools/cache/controller.go:192 +0x36
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:226 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x73aa65?, {0x3068c20, 0xc0008ad380}, 0x1, 0xc0009157a0)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:227 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0008e06b8?, 0x3b9aca00, 0x0, 0x0?, 0x174876e800?)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:204 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:161
k8s.io/client-go/tools/cache.(*controller).Run(0xc0008e0640, 0xc0009157a0)
	k8s.io/client-go@v0.27.0/tools/cache/controller.go:163 +0x385
k8s.io/client-go/tools/cache.(*sharedIndexInformer).Run(0xc000add4a0, 0xc000824c00?)
	k8s.io/client-go@v0.27.0/tools/cache/shared_informer.go:503 +0x542
k8s.io/client-go/informers.(*sharedInformerFactory).Start.func1()
	k8s.io/client-go@v0.27.0/informers/factory.go:150 +0x6b
created by k8s.io/client-go/informers.(*sharedInformerFactory).Start
	k8s.io/client-go@v0.27.0/informers/factory.go:148 +0x21b
E0715 23:48:08.032039       1 runtime.go:79] Observed a panic: &errors.errorString{s:"unable to calculate an index entry for key \"147-185-40-113\" on index \"instanceID\": error mapping node \"147-185-40-113\"'s provider ID \"not:///asdf/asdf\" to instance ID: Invalid format for AWS instance (not:///asdf/asdf)"} (unable to calculate an index entry for key "147-185-40-113" on index "instanceID": error mapping node "147-185-40-113"'s provider ID "not:///asdf/asdf" to instance ID: Invalid format for AWS instance (not:///asdf/asdf))
goroutine 355 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x25e3d20?, 0xc000f38170})
	k8s.io/apimachinery@v0.27.0/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0008cb430?})
	k8s.io/apimachinery@v0.27.0/pkg/util/runtime/runtime.go:49 +0x75
panic({0x25e3d20, 0xc000f38170})
	runtime/panic.go:884 +0x213
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xfffffffe?})
	k8s.io/apimachinery@v0.27.0/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x25e3d20, 0xc000f38170})
	runtime/panic.go:884 +0x213
k8s.io/client-go/tools/cache.(*storeIndex).updateIndices(0xc000b882a0, {0x0, 0x0}, {0x2aa73c0?, 0xc000f2e000}, {0xc000cb37c0, 0xe})
	k8s.io/client-go@v0.27.0/tools/cache/thread_safe_store.go:162 +0x6b3
k8s.io/client-go/tools/cache.(*threadSafeMap).Update(0xc000b24ea0, {0xc000cb37c0, 0xe}, {0x2aa73c0?, 0xc000f2e000})
	k8s.io/client-go@v0.27.0/tools/cache/thread_safe_store.go:226 +0x14b
k8s.io/client-go/tools/cache.(*threadSafeMap).Add(0x2aa73c0?, {0xc000cb37c0?, 0x2aa73c0?}, {0x2aa73c0?, 0xc000f2e000?})
	k8s.io/client-go@v0.27.0/tools/cache/thread_safe_store.go:218 +0x28
k8s.io/client-go/tools/cache.(*cache).Add(0xc000a9aa50, {0x2aa73c0, 0xc000f2e000})
	k8s.io/client-go@v0.27.0/tools/cache/store.go:155 +0xb0
k8s.io/client-go/tools/cache.processDeltas({0x308d1e0, 0xc000add4a0}, {0x309a860, 0xc000a9aa50}, {0xc000cc8600?, 0x1, 0xc000c4d1e0?}, 0x28?)
	k8s.io/client-go@v0.27.0/tools/cache/controller.go:459 +0x1c9
k8s.io/client-go/tools/cache.(*sharedIndexInformer).HandleDeltas(0xc000add4a0, {0x269f960?, 0xc000ede960?}, 0x1?)
	k8s.io/client-go@v0.27.0/tools/cache/shared_informer.go:638 +0x176
k8s.io/client-go/tools/cache.(*DeltaFIFO).Pop(0xc000a7a2c0, 0xc0008d9950)
	k8s.io/client-go@v0.27.0/tools/cache/delta_fifo.go:603 +0x5be
k8s.io/client-go/tools/cache.(*controller).processLoop(0xc0008e0640)
	k8s.io/client-go@v0.27.0/tools/cache/controller.go:192 +0x36
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:226 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x73aa65?, {0x3068c20, 0xc0008ad380}, 0x1, 0xc0009157a0)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:227 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0008e06b8?, 0x3b9aca00, 0x0, 0x0?, 0x174876e800?)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:204 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:161
k8s.io/client-go/tools/cache.(*controller).Run(0xc0008e0640, 0xc0009157a0)
	k8s.io/client-go@v0.27.0/tools/cache/controller.go:163 +0x385
k8s.io/client-go/tools/cache.(*sharedIndexInformer).Run(0xc000add4a0, 0xc000824c00?)
	k8s.io/client-go@v0.27.0/tools/cache/shared_informer.go:503 +0x542
k8s.io/client-go/informers.(*sharedInformerFactory).Start.func1()
	k8s.io/client-go@v0.27.0/informers/factory.go:150 +0x6b
created by k8s.io/client-go/informers.(*sharedInformerFactory).Start
	k8s.io/client-go@v0.27.0/informers/factory.go:148 +0x21b
E0715 23:48:08.032154       1 runtime.go:79] Observed a panic: &errors.errorString{s:"unable to calculate an index entry for key \"147-185-40-113\" on index \"instanceID\": error mapping node \"147-185-40-113\"'s provider ID \"not:///asdf/asdf\" to instance ID: Invalid format for AWS instance (not:///asdf/asdf)"} (unable to calculate an index entry for key "147-185-40-113" on index "instanceID": error mapping node "147-185-40-113"'s provider ID "not:///asdf/asdf" to instance ID: Invalid format for AWS instance (not:///asdf/asdf))
goroutine 355 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x25e3d20?, 0xc000f38170})
	k8s.io/apimachinery@v0.27.0/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000a43440?})
	k8s.io/apimachinery@v0.27.0/pkg/util/runtime/runtime.go:49 +0x75
panic({0x25e3d20, 0xc000f38170})
	runtime/panic.go:884 +0x213
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0008cb430?})
	k8s.io/apimachinery@v0.27.0/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x25e3d20, 0xc000f38170})
	runtime/panic.go:884 +0x213
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xfffffffe?})
	k8s.io/apimachinery@v0.27.0/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x25e3d20, 0xc000f38170})
	runtime/panic.go:884 +0x213
k8s.io/client-go/tools/cache.(*storeIndex).updateIndices(0xc000b882a0, {0x0, 0x0}, {0x2aa73c0?, 0xc000f2e000}, {0xc000cb37c0, 0xe})
	k8s.io/client-go@v0.27.0/tools/cache/thread_safe_store.go:162 +0x6b3
k8s.io/client-go/tools/cache.(*threadSafeMap).Update(0xc000b24ea0, {0xc000cb37c0, 0xe}, {0x2aa73c0?, 0xc000f2e000})
	k8s.io/client-go@v0.27.0/tools/cache/thread_safe_store.go:226 +0x14b
k8s.io/client-go/tools/cache.(*threadSafeMap).Add(0x2aa73c0?, {0xc000cb37c0?, 0x2aa73c0?}, {0x2aa73c0?, 0xc000f2e000?})
	k8s.io/client-go@v0.27.0/tools/cache/thread_safe_store.go:218 +0x28
k8s.io/client-go/tools/cache.(*cache).Add(0xc000a9aa50, {0x2aa73c0, 0xc000f2e000})
	k8s.io/client-go@v0.27.0/tools/cache/store.go:155 +0xb0
k8s.io/client-go/tools/cache.processDeltas({0x308d1e0, 0xc000add4a0}, {0x309a860, 0xc000a9aa50}, {0xc000cc8600?, 0x1, 0xc000c4d1e0?}, 0x28?)
	k8s.io/client-go@v0.27.0/tools/cache/controller.go:459 +0x1c9
k8s.io/client-go/tools/cache.(*sharedIndexInformer).HandleDeltas(0xc000add4a0, {0x269f960?, 0xc000ede960?}, 0x1?)
	k8s.io/client-go@v0.27.0/tools/cache/shared_informer.go:638 +0x176
k8s.io/client-go/tools/cache.(*DeltaFIFO).Pop(0xc000a7a2c0, 0xc0008d9950)
	k8s.io/client-go@v0.27.0/tools/cache/delta_fifo.go:603 +0x5be
k8s.io/client-go/tools/cache.(*controller).processLoop(0xc0008e0640)
	k8s.io/client-go@v0.27.0/tools/cache/controller.go:192 +0x36
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:226 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x73aa65?, {0x3068c20, 0xc0008ad380}, 0x1, 0xc0009157a0)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:227 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0008e06b8?, 0x3b9aca00, 0x0, 0x0?, 0x174876e800?)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:204 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:161
k8s.io/client-go/tools/cache.(*controller).Run(0xc0008e0640, 0xc0009157a0)
	k8s.io/client-go@v0.27.0/tools/cache/controller.go:163 +0x385
k8s.io/client-go/tools/cache.(*sharedIndexInformer).Run(0xc000add4a0, 0xc000824c00?)
	k8s.io/client-go@v0.27.0/tools/cache/shared_informer.go:503 +0x542
k8s.io/client-go/informers.(*sharedInformerFactory).Start.func1()
	k8s.io/client-go@v0.27.0/informers/factory.go:150 +0x6b
created by k8s.io/client-go/informers.(*sharedInformerFactory).Start
	k8s.io/client-go@v0.27.0/informers/factory.go:148 +0x21b
panic: unable to calculate an index entry for key "147-185-40-113" on index "instanceID": error mapping node "147-185-40-113"'s provider ID "not:///asdf/asdf" to instance ID: Invalid format for AWS instance (not:///asdf/asdf) [recovered]
	panic: unable to calculate an index entry for key "147-185-40-113" on index "instanceID": error mapping node "147-185-40-113"'s provider ID "not:///asdf/asdf" to instance ID: Invalid format for AWS instance (not:///asdf/asdf) [recovered]
	panic: unable to calculate an index entry for key "147-185-40-113" on index "instanceID": error mapping node "147-185-40-113"'s provider ID "not:///asdf/asdf" to instance ID: Invalid format for AWS instance (not:///asdf/asdf) [recovered]
	panic: unable to calculate an index entry for key "147-185-40-113" on index "instanceID": error mapping node "147-185-40-113"'s provider ID "not:///asdf/asdf" to instance ID: Invalid format for AWS instance (not:///asdf/asdf)

goroutine 355 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000a43440?})
	k8s.io/apimachinery@v0.27.0/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x25e3d20, 0xc000f38170})
	runtime/panic.go:884 +0x213
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0008cb430?})
	k8s.io/apimachinery@v0.27.0/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x25e3d20, 0xc000f38170})
	runtime/panic.go:884 +0x213
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xfffffffe?})
	k8s.io/apimachinery@v0.27.0/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x25e3d20, 0xc000f38170})
	runtime/panic.go:884 +0x213
k8s.io/client-go/tools/cache.(*storeIndex).updateIndices(0xc000b882a0, {0x0, 0x0}, {0x2aa73c0?, 0xc000f2e000}, {0xc000cb37c0, 0xe})
	k8s.io/client-go@v0.27.0/tools/cache/thread_safe_store.go:162 +0x6b3
k8s.io/client-go/tools/cache.(*threadSafeMap).Update(0xc000b24ea0, {0xc000cb37c0, 0xe}, {0x2aa73c0?, 0xc000f2e000})
	k8s.io/client-go@v0.27.0/tools/cache/thread_safe_store.go:226 +0x14b
k8s.io/client-go/tools/cache.(*threadSafeMap).Add(0x2aa73c0?, {0xc000cb37c0?, 0x2aa73c0?}, {0x2aa73c0?, 0xc000f2e000?})
	k8s.io/client-go@v0.27.0/tools/cache/thread_safe_store.go:218 +0x28
k8s.io/client-go/tools/cache.(*cache).Add(0xc000a9aa50, {0x2aa73c0, 0xc000f2e000})
	k8s.io/client-go@v0.27.0/tools/cache/store.go:155 +0xb0
k8s.io/client-go/tools/cache.processDeltas({0x308d1e0, 0xc000add4a0}, {0x309a860, 0xc000a9aa50}, {0xc000cc8600?, 0x1, 0xc000c4d1e0?}, 0x28?)
	k8s.io/client-go@v0.27.0/tools/cache/controller.go:459 +0x1c9
k8s.io/client-go/tools/cache.(*sharedIndexInformer).HandleDeltas(0xc000add4a0, {0x269f960?, 0xc000ede960?}, 0x1?)
	k8s.io/client-go@v0.27.0/tools/cache/shared_informer.go:638 +0x176
k8s.io/client-go/tools/cache.(*DeltaFIFO).Pop(0xc000a7a2c0, 0xc0008d9950)
	k8s.io/client-go@v0.27.0/tools/cache/delta_fifo.go:603 +0x5be
k8s.io/client-go/tools/cache.(*controller).processLoop(0xc0008e0640)
	k8s.io/client-go@v0.27.0/tools/cache/controller.go:192 +0x36
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:226 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x73aa65?, {0x3068c20, 0xc0008ad380}, 0x1, 0xc0009157a0)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:227 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0008e06b8?, 0x3b9aca00, 0x0, 0x0?, 0x174876e800?)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:204 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
	k8s.io/apimachinery@v0.27.0/pkg/util/wait/backoff.go:161
k8s.io/client-go/tools/cache.(*controller).Run(0xc0008e0640, 0xc0009157a0)
	k8s.io/client-go@v0.27.0/tools/cache/controller.go:163 +0x385
k8s.io/client-go/tools/cache.(*sharedIndexInformer).Run(0xc000add4a0, 0xc000824c00?)
	k8s.io/client-go@v0.27.0/tools/cache/shared_informer.go:503 +0x542
k8s.io/client-go/informers.(*sharedInformerFactory).Start.func1()
	k8s.io/client-go@v0.27.0/informers/factory.go:150 +0x6b
created by k8s.io/client-go/informers.(*sharedInformerFactory).Start
	k8s.io/client-go@v0.27.0/informers/factory.go:148 +0x21b
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
kubectl version
Client Version: v1.32.6+k3s1
Kustomize Version: v5.5.0
Server Version: v1.32.6+k3s1

- Cloud provider or hardware configuration:
AWS and hybrid nodes
- OS (e.g. from /etc/os-release):
ubuntu@ip-192-168-212-11:~$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.5 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.5 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
- Kernel (e.g. `uname -a`):
Linux ip-192-168-212-11 6.8.0-1031-aws #33~22.04.1-Ubuntu SMP Thu Jun 26 14:22:30 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
- Others:

<!-- DO NOT EDIT BELOW THIS LINE -->
/kind bug

--- a/charts/aws-cloud-controller-manager/values.yaml
+++ b/charts/aws-cloud-controller-manager/values.yaml
@@ -6,7 +6,7 @@ args:
 
 image:
     repository: registry.k8s.io/provider-aws/cloud-controller-manager
-    tag: v1.27.1
+    tag: v1.33.0
 
 # Specify image pull secrets
 imagePullSecrets: []

